drivers/shared/executor/executor.go

```diff
@@ -312,7 +312,11 @@ func (e *UniversalExecutor) Launch(command *ExecCommand) (*ProcessState, error)
 		return nil, fmt.Errorf("failed to start command path=%q --- args=%q: %v", path, e.childCmd.Args, err)
 	}
 
-	go e.pidCollector.collectPids(e.processExited, getAllPids)
+	if e.resConCtx.isEmpty() {
+		go e.pidCollector.collectPids(e.processExited, getAllPids)
+	} else {
+		go e.pidCollector.collectPids(e.processExited, e.getAllPids)
```

Comment:
Reusing the same function name to refer to a executor method as well as a package function with different implementations make it a bit harder to follow code.  Apparently, we do that already with libcontainer executor :(.

Also, `l.getAllPids` and `resConCtx.isEmpty` are only defined on linux, so compilation fails on Windows and macOS, e.g. https://ci.appveyor.com/project/hashicorp/nomad/builds/25919504 .  See the pattern of `executor_universal_linux.go`, `executor_windows.go`, and `executor_unix.go` .

I would suggest refactoring this to have a universal executor `getAllPids` method that calls helpers to look up pids (e.g. `getPidsByCgroup` vs `getPidsByScanningProcesses`)?